### ISSUE TITLE
Possible bug in the data collector 

### ISSUE THREAD

**Issue by marcelovmaciel** [JuliaDynamics/Agents.jl#179]
Hi everyone. I have a problem and would like to know if this is a problem in  my code or a bug. I've written a simulation (https://github.com/marcelovmaciel/Pudley/blob/master/Pudley/src/01-basefns.jl) in which I set , "by hand", that the first agent , id = 1, would have an attribute , r,  with value 0. for this agent this attribute will always be zero (while it changes for other agents throughout the simulation).  when I looked the data collected by `step!`  it showed that the agent with id = 1 , in the dataframe column id,  didnt have  r = 0 even at step = 0. Actually, if I set the population size to n, agent n will be the one with r= 0. When I call `id2agent(1,m),` however, it returns me the correct agent. I have modified the hegselmann-krause model that is in the tutorials to see if the same happened and it did (its here https://github.com/marcelovmaciel/mvm-abm-zoo/blob/master/AbmZoo/src/HK.jl). 
( the first model was implemented in Agents v.2.4, while this modified HK was run in Agents#master)

**Comment by Datseris** [comment]
linked example:
```julia

using Agents
using Distributions: mean
using DataVoyager
using BenchmarkTools

mutable struct HKAgent{T<:AbstractFloat} <: AbstractAgent
    id::Int
    old_opinion::T
    new_opinion::T
    r::T
end


function hk_model(; numagents = 100, ϵ = 0.4)
    model = ABM(HKAgent{Float64}, scheduler = fastest, properties = Dict(:ϵ => ϵ))

    o1 = rand()
    add_agent!(model, o1,o1, 0)
    for i = 2:numagents
        o = rand()
        add_agent!(model, o, o, 1 )
    end
    return model
end

get_old_opinion(agent)::Float64 = agent.old_opinion
o(agent) = get_old_opinion(agent)

function xr!(a, m)
    central_agent = id2agent(1, m)
    xᵣₐ = (o(a) - o(central_agent))
    a.r = xᵣₐ
end

function boundfilter(agent, model)
    filter(
        j -> abs(o(agent) - j) < model.properties[:ϵ],
        o.(values(model.agents)),
    )
end

function agent_step!(agent, model)
    agent.new_opinion = mean(boundfilter(agent, model))
end

function updateold(a)
    a.old_opinion = a.new_opinion
    return a
end

function model_step!(model)
    for i in keys(model.agents)
        agent = id2agent(i, model)
        updateold(agent)
        xr!(agent,model)
    end
end

function model_run(; numagents = 100, iterations = 50, ϵ = 0.3)
    model = hk_model(numagents = numagents, ϵ = ϵ)
    println(id2agent(1,model))
    when = 0:5:iterations
    agent_properties = [:new_opinion, :r]
    data = step!(model, agent_step!, model_step!, iterations, agent_properties, when = when)
    return (data)
end



data = model_run(numagents = 5, ϵ = 0.0001)
filter(x -> x.step ==
       0 , data )
```

**Comment by Datseris** [comment]
Yeah, this is definitely a bug (but I can't fix it). Unless you want to make your hands dirty and go into the source code of collect data, which is here: https://github.com/JuliaDynamics/Agents.jl/blob/master/src/simulations/data_collector.jl   , this will take a while to get resolved.

**Comment by Datseris** [comment]
Actually found the bug: https://github.com/JuliaDynamics/Agents.jl/blob/master/src/simulations/data_collector.jl#L77 

Can you please try this with the `sort` function deleted? (if you are not familiar, `dev Agents`, go to the source `~/.julia/dev/Agents`, go to the file, change this line, then load agents and try.

**Comment by Datseris** [comment]
I don't know why @kavir1698 decided to sort the ids there. (Also, all of these begin statements are unecessary, they should be deleted)

**Comment by Datseris** [comment]
@marcelovmaciel does this fix the problem for you? In your real case?

**Comment by Datseris** [comment]
It does. I tested and fixed it in #186

**Comment by marcelovmaciel** [comment]
@Datseris i only saw the message now. Thanks !!!!!
